pkg/tools/tools.go

```diff
@@ -145,32 +145,23 @@ func ParseKVStringToMap(input, sep, kvsep string) map[string]string {
 
 // initJaeger returns an instance of Jaeger Tracer that samples 100% of traces and logs all spans to stdout.
 func InitJaeger(service string) (opentracing.Tracer, io.Closer) {
-	jaegerHost := os.Getenv("JAEGER_SERVICE_HOST")
-	jaegerPort := os.Getenv("JAEGER_SERVICE_PORT_JAEGER")
-	jaegerHostPort := fmt.Sprintf("%s:%s", jaegerHost, jaegerPort)
-
-	logrus.Infof("Using Jaeger host/port: %s", jaegerHostPort)
-
-	cfg := &config.Configuration{
-		Sampler: &config.SamplerConfig{
-			Type:  "const",
-			Param: 1,
-		},
-		Reporter: &config.ReporterConfig{
-			LogSpans:           true,
-			LocalAgentHostPort: jaegerHostPort,
-		},
+	cfg, err := config.FromEnv()
+	if err != nil {
+		panic(fmt.Sprintf("ERROR: cannot create Jaeger configuration: %v\n", err))
 	}
 
+	cfg.Sampler.Type = "const"
+	cfg.Sampler.Param = 1
+	cfg.Reporter.LogSpans = true
+
 	hostname, err := os.Hostname()
```

Comment:
maybe we should only set cfg.ServiceName in here if it hasn't already been set by the config.FromEnv var `JAEGER_SERVICE_NAME`

Reply:
Fixed. What about other config parameters we set? They also may be not empty since they can be set via envs.

Reply:
Good point.  I agree that it's better to detect whether they're set the from env and only default to these values if unset.

Reply:
Done